### PR TITLE
Snow integrated temperature

### PR DESCRIPTION
**Pull request by lvankampenhout**
adding diagnostic variable SNOTXMASS for CMIP6

List of changes: 
- added SNOTXMASS 
- added SNOTXMASS_ICE (inactive)
- added SNOWICE_ICE (inactive)
- added SNOWLIQ_ICE (inactive)
- added c2l scale types for SNOWICE, SNOWLIQ

Example Python code that processes the new diagnostic variable to compute snow internal temperature can be found here:

https://github.com/lvankampenhout/Jupyter_notebooks/blob/master/Snow%20internal%20temperature%20in%20CLM.ipynb

### CODE REVIEW

File: src/biogeophys/TemperatureType.F90 [hunk 1]
```diff
@@ -410,6 +412,16 @@ subroutine InitHistory(this, bounds, is_simple_buildtemp, is_prog_buildtemp )
          avgflag='A', long_name='soil temperature in top 10cm of soil', &
          ptr_col=this%t_soi10cm_col, set_urb=spval)
 
+    this%t_sno_mul_mss_col(begc:endc) = spval
+    call hist_addfld1d (fname='SNOTXMASS',  units='K kg/m2', &
+         avgflag='A', long_name='snow temperature times layer mass, layer sum', &
```
Comment (billsacks) [inline src/biogeophys/TemperatureType.F90]: This and the related _ICE variable should probably add in the long_name something like this:

```Fortran
         avgflag='A', &
         long_name='snow temperature times layer mass, layer sum; '// &
         'to get mass-weighted temperature, divide by (SNOWICE+SNOWLIQ)', &
```

**Comment by billsacks** [comment]
I will merge this onto a branch containing other changes needed for #271 , and bring this branch to master next week.

File: src/biogeophys/TemperatureType.F90 (hunk 1 above)
Comment (lvankampenhout) [reply src/biogeophys/TemperatureType.F90]: Great suggestion and I agree with your wording. Thanks

**Comment by lvankampenhout** [comment]
Hi Bill, I'm a little confused by the github interface here, as I can't seem to find back your original comment. What I wanted to say is that I've changed the long_names to be a bit more descriptive like you suggested. It's in my branch and therefore also part of this pull request (happens automatically)

**Comment by billsacks** [comment]
> Hi Bill, I'm a little confused by the github interface here, as I can't seem to find back your original comment. What I wanted to say is that I've changed the long_names to be a bit more descriptive like you suggested. It's in my branch and therefore also part of this pull request (happens automatically)

Thanks. For future reference: When you change a line of code that has a comment, github hides the comment. If you search for "show outdated" on this page you'll see a button that lets you expand the comment.

**Pull request merged**